theano/tensor/extra_ops.py

```diff
@@ -68,6 +70,164 @@ def c_code_cache_version(self):
 cpu_contiguous = CpuContiguous()
 
 
+class SearchsortedOp(theano.Op):
+    """Wrapper of numpy.searchsorted.
+
+    For full documentation, see :func:`searchsorted`.
+
+    See Also
+    --------
+    searchsorted : numpy-like function to use the SearchsortedOp
+
+    """
+
+    params_type = Generic()
+    __props__ = ("side", )
+
+    def __init__(self, side='left'):
+        self.side = side
```

Comment:
Maybe I should let numpy's exception to be raised. I will change the c_code implementation to allow this.